### PR TITLE
refactor: removed async visit methods

### PR DESCRIPTION
**Pull request by ronimizy**

### CODE REVIEW

File: Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlVisitor.cs [hunk 1]
```diff
@@ -14,53 +14,48 @@ public ClosedXmlVisitor(IXLWorksheet worksheet, Index index, Style style = defau
         _worksheet = worksheet;
     }
 
-    protected override Task StyleRangeAsync(Style style, IndexRange range, CancellationToken cancellationToken)
+    public override Task ApplyChangesAsync(CancellationToken cancellationToken = default)
+        => Task.CompletedTask;
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlVisitor.cs:18]: I thought ApplyChangesAsync will call `workbook.SaveAs` or `workbook.Save`, like GoogleSheetVisitor's ApplyChangesAsync, that will send google api requests. I think it's kinda misguiding that you're Applying Changes, but nothing's happening and you have to manually save workbook afterwards

File: Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlVisitor.cs [hunk 2]
```diff
@@ -14,53 +14,48 @@ public ClosedXmlVisitor(IXLWorksheet worksheet, Index index, Style style = defau
...
         _worksheet.Rows(from, upTo - 1).Height = height;
-        return Task.CompletedTask;
     }
 
-    protected override Task SetColumnWidthAsync(int from, int upTo, int width, CancellationToken cancellationToken)
+    protected override void SetColumnWidth(int from, int upTo, int width)
     {
         _worksheet.Columns(from, upTo - 1).Width = width;
-        return Task.CompletedTask;
     }
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlVisitor.cs:60]: Every method here can have an expression body, but I wont insist

File: Source/FluentSpreadsheets/ComponentImplementations/VStackComponent.cs [hunk 3]
```diff
@@ -47,8 +47,8 @@ public VStackComponent(IReadOnlyCollection<IComponent> components, int width)
     
     public IReadOnlyCollection<IComponent> Components { get; }
 
-    public Task AcceptAsync(IComponentVisitor visitor, CancellationToken cancellationToken)
+    public void Accept(IComponentVisitor visitor)
     {
-        return visitor.VisitAsync(this, cancellationToken);
+         visitor.Visit(this);
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets/ComponentImplementations/VStackComponent.cs:52]:
```suggestion
        visitor.Visit(this);
```

**Review by Mihinator3000 — CHANGES_REQUESTED** [review]

File: Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
+namespace FluentSpreadsheets.Visitors;
+
+public class ComponentVisitor<THandler> : IComponentVisitor where THandler : IComponentVisitorHandler
+{
+    public readonly THandler Handler;
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs:5]: ?

File: Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
+namespace FluentSpreadsheets.Visitors;
+
+public class ComponentVisitor<THandler> : IComponentVisitor where THandler : IComponentVisitorHandler
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs:3]:
```suggestion
public class ComponentVisitor<THandler> : IComponentVisitor
    where THandler : IComponentVisitorHandler
```

File: Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlHandler.cs [hunk 6]
```diff
@@ -4,20 +4,16 @@
...
 namespace FluentSpreadsheets.ClosedXML.Visitors;
 
-public class ClosedXmlVisitor : ComponentVisitorBase
+internal readonly struct ClosedXmlHandler : IComponentVisitorHandler
 {
     private readonly IXLWorksheet _worksheet;
 
-    public ClosedXmlVisitor(IXLWorksheet worksheet, Index index, Style style = default)
-        : base(index, style)
+    public ClosedXmlHandler(IXLWorksheet worksheet)
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlHandler.cs:11]: record struct?

File: Source/FluentSpreadsheets.ClosedXML/Rendering/ClosedXmlComponentRenderer.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+using FluentSpreadsheets.Visitors;
+
+namespace FluentSpreadsheets.ClosedXML.Rendering;
+
+public class ClosedXmlComponentRenderer : IComponentRenderer<ClosedXmlRenderCommand>
+{
+    public Task RenderAsync(ClosedXmlRenderCommand command, CancellationToken cancellationToken = default)
+    {
+        var handler = new ClosedXmlHandler(command.Worksheet);
+        var visitor = new ComponentVisitor<ClosedXmlHandler>(new Index(1, 1), handler);
```
Comment (Mihinator3000) [inline Source/FluentSpreadsheets.ClosedXML/Rendering/ClosedXmlComponentRenderer.cs:12]: Why exactly are we passing new Index every time?

File: Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlHandler.cs (hunk 6 above)
Comment (ronimizy) [reply Source/FluentSpreadsheets.ClosedXML/Visitors/ClosedXmlHandler.cs:11]: no

File: Source/FluentSpreadsheets.ClosedXML/Rendering/ClosedXmlComponentRenderer.cs (hunk 7 above)
Comment (ronimizy) [reply Source/FluentSpreadsheets.ClosedXML/Rendering/ClosedXmlComponentRenderer.cs:12]: it is parametrisation. component visitor lets you customise a starting point of rendered component. as index is a struct, there is no difference between storing it in a field and pass to visitor's constructor, define a constructor with default index, or the way it is now.

File: Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs (hunk 5 above)
Comment (ronimizy) [reply Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs:3]: single generic parameter constraint looks dangling on next line, so it is better to keep it on the same line.

Comment (Mihinator3000) [reply Source/FluentSpreadsheets/Visitors/ComponentVisitor.cs:3]: Okay, but imho, its just hard to read in single line

File: Source/FluentSpreadsheets.ClosedXML/Rendering/ClosedXmlComponentRenderer.cs (hunk 7 above)
Comment (Mihinator3000) [reply Source/FluentSpreadsheets.ClosedXML/Rendering/ClosedXmlComponentRenderer.cs:12]: K, I just think it'll be cleaner with one more constructor

**Review by Mihinator3000 — APPROVED** [review]

**Pull request merged**